nupm/utils/registry.nu

```diff
@@ -0,0 +1,77 @@
+# Utilities related to nupm registries
+
+# Search for a package in a registry
+export def search-package [
+    package: string  # Name of the package
+    --registry: string  # Which registry to use
+    --version: any  # Package version to install (string or null)
+    --exact-match  # Searched package name must match exactly
+] -> table {
+    let registries = if (not ($registry | is-empty)) and ($registry in $env.NUPM_REGISTRIES) {
+        # If $registry is a valid column in $env.NUPM_REGISTRIES, use that
+        { $registry : ($env.NUPM_REGISTRIES | get $registry) }
+    } else if (not ($registry | is-empty)) and ($registry | path exists) {
+        # If $registry is a path, use that
+        let reg_name = $registry | path parse | get stem
+        { $reg_name: $registry }
+    } else {
+        # Otherwise use $env.NUPM_REGISTRIES
+        $env.NUPM_REGISTRIES
+    }
+
+    let name_matcher: closure = if $exact_match {
+        {|row| $row.name == $package }
+    } else {
+        {|row| $package in $row.name }
+    }
+
+    # Collect all registries matching the package and all matching packages
+    let regs = $registries
+        | items {|name, path|
+            # Open registry (online or offline)
+            let registry = if ($path | path type) == file {
+                open $path
+            } else {
+                try {
+                    let reg = http get $path
+
+                    if local in $reg {
+                        throw-error ("Can't have local packages in online registry"
+                            + $" '($path)'.")
+                    }
+
+                    $reg
+                } catch {
+                    throw-error $"Cannot open '($path)' as a file or URL."
```

Comment:
i'm confused, where are `throw-error` and `check-cols` imported from in this module ?? :open_mouth: